### PR TITLE
1082 group is not updated after delete user

### PR DESCRIPTION
**Pull request by tcnichol**
This is draft, I think I found a fix but my pre commit is still acting up.

### CODE REVIEW

**Comment by tcnichol** [comment]
I think something is messed up here. I only wanted changes in Group component, I reverted changes in the other one, but it still shows changes.
Draft till I figure this out.

**Comment by longshuicy** [comment]
I switched this to draft according to your comments; if this is ready feel free to switch it back. Thanks!

**Comment by ddey2** [comment]
@tcnichol I see this

Screenshot 2024-06-28 at 12 26 33 PM

**Comment by ddey2** [comment]
@tcnichol You might want to merge main to resolve the above error. 

Also, it doesn't work. I deleted user but it doesn't update it on the page unless you refresh it.

**Review by ddey2 — CHANGES_REQUESTED** [review]
@tcnichol  You might want to merge main to resolve the above error.

Also, it doesn't work. I deleted user but it doesn't update it on the page unless you refresh it.

**Comment by tcnichol** [comment]
> @tcnichol You might want to merge main to resolve the above error.
> 
> Also, it doesn't work. I deleted user but it doesn't update it on the page unless you refresh it.

Can you try to delete the node_modules file and then do npm install again? When I switched to another branch and back, it wasn't working until I did npm install again.

File: frontend/src/components/groups/Group.tsx [hunk 1]
```diff
@@ -1,103 +1,82 @@
...
-		},
-	];
+	}, [groupAbout]);
 
 	return (
 		<Layout>
-			{/*Error Message dialogue*/}
-			<ErrorModal errorOpen={errorOpen} setErrorOpen={setErrorOpen} />
-			{/*breadcrumb*/}
-			<Grid container>
```
Comment (longshuicy) [inline frontend/src/components/groups/Group.tsx:74]: any reason you are removing the breadcrumb?

File: frontend/src/components/groups/Group.tsx [hunk 2]
```diff
@@ -1,103 +1,82 @@
...
+			<ActionModal
+				actionOpen={deleteGroupConfirmOpen}
+				actionTitle="Are you sure?"
+				actionText="Do you really want to delete this group? This process cannot be undone."
+				actionBtnName="Delete"
+				handleActionBtnClick={() => {
+					groupDeleted(groupId);
+					setDeleteGroupConfirmOpen(false);
+					// Go to Explore page
+					history("/");
```
Comment (longshuicy) [inline frontend/src/components/groups/Group.tsx:83]: Is it ok to go to /groups page instead?

File: frontend/src/components/groups/Group.tsx (hunk 1 above)
Comment (tcnichol) [reply frontend/src/components/groups/Group.tsx:74]: No, this must have been a mistake. I have put it back in.

File: frontend/src/components/groups/Group.tsx [hunk 3]
```diff
@@ -149,16 +143,25 @@ export function Group() {
 								setAddMemberModalOpen(true);
 							}}
 							endIcon={<PersonAddAlt1Icon />}
-							sx={{ marginRight: "0.5em", width: "auto" }}
 						>
 							Add Member
 						</Button>
 					</AuthWrapper>
-					<AuthWrapper currRole={role} allowedRoles={["owner", "editor"]}>
-						<EditMenu setDeleteGroupConfirmOpen={setDeleteGroupConfirmOpen} />
```
Comment (longshuicy) [inline frontend/src/components/groups/Group.tsx:158]: Any reason you are get rid of the edit menu which includes "edit group names etc"?

**Review by longshuicy — CHANGES_REQUESTED** [review]
This PR works but it seems to get rid of a few existing functionality. 
Also a little explanation on what's the culprit of the issue and what's the fix would be greatly appreciated!

File: frontend/src/components/groups/Group.tsx (hunk 3 above)
Comment (tcnichol) [reply frontend/src/components/groups/Group.tsx:158]: I think this was a mistake somehow. I will fix this and check to see what else is missing from main.

**Comment by tcnichol** [comment]
@longshuicy 

I went back and reverted all of these files to main, and then just made the change necessary to make this work. I'm not sure why it got changed so much.

The basic thing was the useEffect hook needed to depend on groupAbout, so it would populate the group members correctly when groupAbout changed after a user was deleted from the group. Now it updates right for delete and add, and that should be the only change.

**Comment by longshuicy** [comment]
Thanks for the explanation. Deletion works well now.

However when I try to edit the role of any user, it wouldn't allow me to select any of those selection
image

**Review by longshuicy — APPROVED** [review]

**Pull request merged**